src/serac/numerics/equation_solver.hpp

```diff
@@ -169,7 +175,7 @@ class EquationSolver : public mfem::Solver {
   /**
    * @brief The optional nonlinear Newton-Raphson solver object
    */
-  std::unique_ptr<mfem::NewtonSolver> nonlin_solver_;
+  std::variant<std::unique_ptr<mfem::NewtonSolver>, mfem::NewtonSolver*> nonlin_solver_;
```

Comment:
this also looks unusual, are the KINSOL solver options all derived from `mfem::NewtonSolver`?